app/src/apdu_handler.c

```diff
@@ -212,6 +216,17 @@ __Z_INLINE void handleSign(volatile uint32_t *flags, volatile uint32_t *tx, uint
         THROW(APDU_CODE_DATA_INVALID);
     }
 
+#ifdef HAVE_SWAP
+    if (G_swap_state.called_from_swap && G_swap_state.should_exit && error_msg == NULL) {
```

Comment:
I think we need to stop regardless of `error_msg`, though I'm not sure in which case it happens
several possibilities depending on logic
- if error rapdu has been sent we can just call `finalize_exchange_sign_transaction(false)`
- otherwise we send it and `finalize_exchange_sign_transaction(false)`